include/swift/AST/Expr.h

```diff
@@ -1035,11 +1036,15 @@ class MagicIdentifierLiteralExpr : public BuiltinLiteralExpr {
 
 private:
   SourceLoc Loc;
+  ArgumentList *ArgList;
+  DefaultArgumentExpr *DefaultArgument;
 
 public:
-  MagicIdentifierLiteralExpr(Kind kind, SourceLoc loc, bool implicit = false)
+  MagicIdentifierLiteralExpr(Kind kind, SourceLoc loc, ArgumentList *argList,
```

Comment:
TODO: refactor the new fields into a new subtype of `MagicIdentifierLiteralExpr` so that they don't bloat memory usage of existing magic identifiers (assuming it's a concern?)